### PR TITLE
aws-pcluster stacks: Use only static spack.yaml

### PR DESCRIPTION
**Pull request by stephenmsachs**
Instead of creating compilers.yaml as needed use a static compilers.yaml for the
aws-pcluster build caches. These entries need to adapted once the container
image changes, but can be used as static objects within spack to enable testing
changes in compilers.yaml.

Also compiler:environment:prepend_path:PATH entry has been removed and set into
$PATH directly.

-------

### CODE REVIEW

**Comment by stephenmsachs** [comment]
Also move the `packages.yaml` directly into `spack.yaml` such that there is one single static configuration file for each stack.

**Comment by stephenmsachs** [comment]
CI works, stil lneed to clean this up.

File: share/spack/gitlab/cloud_pipelines/scripts/pcluster/setup-pcluster.sh [hunk 1]
```diff
@@ -108,5 +108,5 @@ install_compilers() {
 }
 
 set_pcluster_defaults
-install_compilers
-patch_compilers_yaml
+# install_compilers
```
Comment (haampie) [inline share/spack/gitlab/cloud_pipelines/scripts/pcluster/setup-pcluster.sh]: can this file be deleted?

Comment (stephenmsachs) [reply share/spack/gitlab/cloud_pipelines/scripts/pcluster/setup-pcluster.sh]: Yes, that's part of the cleanup I still need to do.

**Comment by stephenmsachs** [comment]
Pipeline https://gitlab.spack.io/spack/spack/-/pipelines/910916 compiled new version of `palace`, i.e. C++ code with multipel dependencies using `%oneapi`. Thus showing that the correct `ld` and `libstdc++` are found. 

I'll remove the new `palace` entry and then this is ready for merge.

**Review by haampie — APPROVED** [review]

**Pull request merged**